src/server.cpp

```diff
@@ -39,6 +39,57 @@ namespace quicr {
 
     void Server::ResolveSubscribe(ConnectionHandle, uint64_t, const SubscribeResponse&) {}
 
+    void Server::BindPublisherTrack(TransportConnId conn_id,
+                                    uint64_t subscribe_id,
+                                    const std::shared_ptr<PublishTrackHandler>& track_handler)
+    {
+        // Generate track alias
+        const auto& tfn = track_handler->GetFullTrackName();
+
+        // Track hash is the track alias for now.
+        auto th = TrackHash(tfn);
+
+        track_handler->SetTrackAlias(th.track_fullname_hash);
+
+        SPDLOG_LOGGER_INFO(
+          logger_, "Bind subscribe track handler conn_id: {0} hash: {1}", conn_id, th.track_fullname_hash);
+
+        std::lock_guard<std::mutex> _(state_mutex_);
+        auto conn_it = connections_.find(conn_id);
+        if (conn_it == connections_.end()) {
+            SPDLOG_LOGGER_ERROR(logger_, "Subscribe track conn_id: {0} does not exist.", conn_id);
+            return;
+        }
+
+        track_handler->SetSubscribeId(subscribe_id);
+
+        track_handler->connection_handle_ = conn_id;
+
+        track_handler->publish_data_ctx_id_ =
+          quic_transport_->CreateDataContext(conn_id,
+                                             track_handler->default_track_mode_ == TrackMode::kDatagram ? false : true,
+                                             track_handler->default_priority_,
+                                             false);
+
+        // Setup the function for the track handler to use to send objects with thread safety
+        track_handler->publish_object_func_ =
+          [&, track_handler = track_handler, subscribe_id = track_handler->GetSubscribeId()](
```

Comment:
I think you can simply write `track_handler` to copy it, but this notation doesn't hurt.

Reply:
Yeah, we can look at changing that later.